Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/RecordNavPage.xaml.cs

```diff
@@ -40,197 +27,289 @@ namespace Ginger.BusinessFlowsLibNew.AddActionMenu
     /// </summary>
     public partial class RecordNavPage : Page
     {
-        bool isRecording = false;
+        public bool IsRecording = false;
         IWindowExplorer mWindowExplorerDriver;
         private Activity mActParentActivity = null;
         Context mContext;
+        RecordingManager RecordingMngr;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="context"></param>
         public RecordNavPage(Context context)
         {
             InitializeComponent();
             mContext = context;
             xWinGridUC.mContext = mContext;
+            xWinGridUC.WindowsComboBox.SelectionChanged += WindowsComboBox_SelectionChanged;
+
+            InitMethods();
+            context.PropertyChanged += Context_PropertyChanged;
         }
 
-        private void RecordingButton_Click(object sender, RoutedEventArgs e)
+        private void InitMethods()
+        {
+            SetPOMControlsVisibility(false);
+            SetControlsDefault();
+            SetMultiplePropertiesGridView();
+            SetDefault(mContext);
+        }
+
+        private void Context_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if(e != null && e.PropertyName == nameof(BusinessFlow))
+            {
+                mContext = (Context)sender;
+                xWinGridUC.mContext = mContext;
+                InitMethods();
+                if (IsRecording)
+                {
+                    IsRecording = !IsRecording;
+                    StopRecording();
+                    SetRecordingButtonText();
+                }
+            }
+        }
+
+        /// <summary>
+        /// This method will set the page to default
+        /// </summary>
+        /// <param name="context"></param>
+        public void SetDefault(Context context)
+        {
```

Comment:
should be called based on Context.PropertyChanged of BF/Bf.currentActivity.

what about check if the Agent supporting recording at all?

Reply:
on context.propertychanged also it is refreshed "Context_PropertyChanged"